### PR TITLE
[IDP-1845] Add documentation

### PR DESCRIPTION
**Pull request by tjosepo**
## Description of changes

Adds documentation to project using Retype.

### CODE REVIEW

File: docs/src/configuration.md
Comment (heqianwang) [inline docs/src/configuration.md:1]: Would it be worthwhile to provide an example of a config file with all of the fields present at the bottom? That may help provide even more visual help for users

File: docs/src/configuration.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- **Type:** `string`
+- **Default:** `"./dist"`
+
+Output directory path.
+
+### watch
+- **Type:** `boolean`
+- **Default:** `false`
+
+Enable watch mode.
```
Comment (heqianwang) [inline docs/src/configuration.md:50]: I see you elaborated a bit over watch mode below, but could be worth it to have a bit more details here as well

File: docs/src/configuration.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+### input
+- **Type:** `string`
+
+Path to the OpenAPI schema file. Can be a local path or URL to a remote file.
+
+### outdir
+- **Type:** `string`
+- **Default:** `"./dist"`
+
+Output directory path.
```
Comment (heqianwang) [inline docs/src/configuration.md:44]: worth it to elaborate a bit more on output of Types/Clients or other files?

File: docs/src/configuration.md
Comment (heqianwang) [reply docs/src/configuration.md:1]: Ok I see that it is mostly covereed by the Getting-Started page; that is the entry point of the documentation right?

**Review by heqianwang — DISMISSED** [review]

File: .github/workflows/publish-docs.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    permissions:
+      pages: write      # to deploy to Pages
+      id-token: write   # to verify the deployment originates from an appropriate source
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup project
+        uses: ./.github/actions/setup
```
[the remark this answers is not in the document]
Comment (tjosepo) [reply .github/workflows/publish-docs.yml:22]: I think I had trust issues when I wrote this code. Compared it to what I found on https://github.com/gsoft-inc/wl-chromado/blob/main/.github/workflows/retype-action.yml, and turns out It's not. I'll remove it.

File: docs/src/configuration.md
Comment (tjosepo) [reply docs/src/configuration.md:1]: I don't mind doing it. I'm a bit concerned that the number of options might grow too big to include all of the fields, but I think the "basic ones" should be included.

Comment (tjosepo) [reply docs/src/configuration.md:1]: Decided against it since it was included in the Getting Started page, and it felt kind of redundant.

**Review by heqianwang — APPROVED** [review]

**Pull request merged**